### DISCUSSION TITLE
Error when using iSCSI frontend type in LXC Container

### DISCUSSION THREAD

**Discussion by manju-rn** [longhorn/longhorn#8100] [Q&A] [answered]
Here is my Setup:
1. I have created a LXC container with Ubuntu:22.04 image and `security.nesting=true` is set. 
2. Installed microk8s and then longhorn via the kubectl (I had to update the env vars in longhorn-driver-deployer for proper deployment)
```
- name: KUBELET_ROOT_DIR
  value: /var/snap/microk8s/common/var/lib/kubelet
```
3. Created a volume with iSCSI FrontEnd and created respective PV and PVC from the UI
4. Created a workload - Postgres Stateful Pod to use the PVC. However, the Postgres reports the below error - this error i believe is coming from Longhorn
`  Warning  FailedAttachVolume  5s (x7 over 40s)  attachdetach-controller  AttachVolume.Attach failed for volume "pg-scsi-vol" : rpc error: code = InvalidArgument desc = volume pg-scsi-vol invalid frontend type iscsi`

As I understand, I cannot use the **Block Devices** in LXC containers, since when I create a Volume with Block Device (in Frontend), the volume and PV / PVC creation is fine, but when attached to Node / workload, the longhorn instance manager gives following error about incorrect file extents
`time="2024-03-04T02:17:59Z" level=warning msg="pg-vol2-r-f619ee23: time=\"2024-03-04T02:17:58Z\" level=fatal msg=\"Error running start replica command\" func=cmd.ReplicaCmd.func1 file=\"replica.go:88\" error=\"file extent is unsupported: operation not supported\"" func="controller.(*InstanceHandler).printInstanceLogs" file="instance_handler.go:467"`

**Comment by derekbit** [comment]
Can you check if the [thread](https://github.com/longhorn/longhorn/issues/4563) is helpful?

**Reply by manju-rn** [reply]
No, it does not help. Looks like in v1.3.1 the iSCSI attachment used to work fine, but in 1.6.0 a volume with iSCSI attachment in LXC container cannot be attached to Node or workload.
I am not sure if any spcific settings are required form LXD to be set in LXC container instance

**Reply by derekbit** [reply]
OK. I missed the error `file extent is unsupported`. What's your underlying filesystem?

**Reply by manju-rn** [reply, marked as answer]
> OK. I missed the error `file extent is unsupported`. What's your underlying filesystem?

Ah i see where you are getting at. Its ZFS.

But when i create VM (from Lxd) i use the same underlying FS, wonder how it works there?
